clients/cobol-lsp-vscode-extension/src/__tests__/commands/RunAnalysisCLITest.spec.ts

```diff
@@ -0,0 +1,283 @@
+    showInformationMessage: jest
+      .fn()
+      .mockImplementation((message: string) => Promise.resolve(message)),
+    showQuickPick: jest.fn(),
+    onDidChangeActiveTextEditor: jest.fn(),
+    createQuickPick: jest
+      .fn()
+      .mockReturnValue({ onDidChangeSelection: jest.fn(), show: jest.fn() }),
+    createOutputChannel: jest.fn().mockReturnValue({
+      appendLine: jest.fn(),
+    }),
+    terminals: {
+      find: jest.fn(),
+    },
+    createTerminal: jest.fn().mockReturnValue({
+      sendText: jest.fn(),
+      show: jest.fn(),
+    }),
+  },
+  workspace: {
+    getConfiguration: jest.fn().mockReturnValue({
+      get: jest.fn().mockReturnValue(9),
+    }),
+    getWorkspaceFolder: jest.fn().mockReturnValue({ name: "workspaceFolder1" }),
+    onDidChangeConfiguration: jest
+      .fn()
+      .mockReturnValue("onDidChangeConfiguration"),
+    workspaceFolders: [{ uri: { fsPath: "ws-path" } } as any],
+    fs: {
+      createDirectory: jest.fn(),
+      readDirectory: jest.fn().mockReturnValue([]),
+      stat: jest.fn().mockReturnValue(2),
+      writeFile: jest.fn(),
+    },
+  },
+  Uri: {
+    file: jest.fn().mockReturnValue("workspaceFolder2"),
+    parse: jest.fn().mockReturnValue({ path: "/storagePath" }),
+    joinPath: jest.fn().mockImplementation((inputUri: vscode.Uri, path) => {
+      return {
+        path: inputUri.path + path,
+        fsPath: inputUri.fsPath + path,
+      };
+    }),
+  },
+}));
+
+const context: any = {
+  extensionUri: { fsPath: "/test" },
+  globalStorageUri: { fsPath: "/storagePath" },
+  subscriptions: [],
+};
+
+describe("Test Analysis CLI command functionality", () => {
+  test("Cobol Analysis - Java", async () => {
+    const testAnalysis = new RunAnalysis(
+      vscode.Uri.parse(context.globalStorageUri),
+      vscode.Uri.parse(context.extensionUri),
+    );
+    vscode.window.showQuickPick = jest.fn().mockReturnValue("Java");
+
+    const runCobolAnalysisCommandSpy = jest.spyOn(
+      testAnalysis as any,
+      "runCobolAnalysisCommand",
+    );
+    const getCurrentFileLocationSpy = jest.spyOn(
+      testAnalysis as any,
+      "getCurrentFileLocation",
+    );
+    const buildJavaCommandSpy = jest.spyOn(
+      testAnalysis as any,
+      "buildJavaCommand",
+    );
+    const buildNativeCommandSpy = jest.spyOn(
+      testAnalysis as any,
+      "buildNativeCommand",
+    );
+    const buildAnalysisCommandPortionSpy = jest.spyOn(
+      testAnalysis as any,
+      "buildAnalysisCommandPortion",
+    );
+    const sendToTerminalSpy = jest.spyOn(testAnalysis as any, "sendToTerminal");
+
+    await testAnalysis.runCobolAnalysisCommand();
+
+    expect(runCobolAnalysisCommandSpy).toHaveBeenCalled();
+    expect(getCurrentFileLocationSpy).toHaveBeenCalled();
+    expect(buildJavaCommandSpy).toHaveBeenCalledWith("/storagePath");
+    expect(buildJavaCommandSpy).toHaveReturnedWith(
+      'java -jar "undefined/server/jar/server.jar" analysis -s "/storagePath" -cf=.',
```

Comment:
The `undefined` in the path suggests that something is wrong here.

Reply:
Resolved in latest commit.